### PR TITLE
Feature investor profile setup

### PR DESCRIPTION
**Pull request by josemak25**
### What does this PR do?
Add CR on API endpoints

#### Description of Task to be completed?

Have the following endpoints working

`GET /api/v1/private/categories`

`POST /api/v1/private/category`

`GET /api/v1/private/user-categories/:id`

`GET /api/v1/private/user-category`

`POST /api/v1/private/user-categories`

### CODE REVIEW

File: api/controllers/CategoryController.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+      return res.json(sendResponse(httpStatus.OK, 'success!', category, null));
+    } catch (err) {
+      next(err);
+    }
+  };
+
+  const create = async (req, res, next) => {
+    try {
+      const category = await CategoryQuery.create({ ...req.body });
```
Comment (Nathan-E) [inline api/controllers/CategoryController.js]: Use object destructing to get the required fields from the body before passing it to the create function...

File: api/middlewares/IsAdmin.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = (req, res, next) => {
+  if (!req.token) {
+    return res.status(401).json({ msg: 'No Authorization was found' });
```
Comment (Nathan-E) [inline api/middlewares/IsAdmin.js]: Change this to the agreed response format

File: api/middlewares/IsAdmin.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  }
+
+  const { user_type } = req.token;
+
+  if (user_type !== 'admin') {
+    return res.status(401).json({ msg: 'You are not Authorized to perform this operation' });
```
Comment (Nathan-E) [inline api/middlewares/IsAdmin.js]: Same here

File: api/middlewares/IsInvestor.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module.exports = (req, res, next) => {
+  const { user_type } = req.token;
+
+  if (user_type !== 'investor') {
+    return res.status(401).json({ msg: "You can't perform this operation" });
```
Comment (Nathan-E) [inline api/middlewares/IsInvestor.js]: Here too

File: api/queries/category.queries.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  create(payload) {
+    return this.Model.create(payload, { individualHooks: true });
+  }
+
+  findAll() {
+    return this.Model.findAll();
+  }
+
+  findByCategory(categories) {
```
Comment (Nathan-E) [inline api/queries/category.queries.js]: Don't you think the argument  'categories' is misleading...when your method is findByCategory

Comment (Nathan-E) [reply api/queries/category.queries.js]: I suggest the argument should be named category...
What do you think?

File: api/queries/user.category.queries.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    this.Model = Model;
+  }
+
+  create({ user_id, categories }) {
+    const payload = categories.map(category => ({ ...category, user_id }));
+    return this.Model.bulkCreate(payload, { individualHooks: true });
+  }
+
+  findByUserId(user_id) {
+    const QUERY = `
```
Comment (Nathan-E) [inline api/queries/user.category.queries.js]: Good Job

File: test/controllers/CategoryController.test.js [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+  const admin = await User.create({
+    name: 'Martin Luke',
+    email: 'martin@mail.com',
+    password: 'securepassword',
+    password2: 'securepassword',
+    phone: '09057373',
+    user_type: 'admin'
+  });
+
+  const res = await request(api)
```
Comment (Nathan-E) [inline test/controllers/CategoryController.test.js]: Use the token service to generate the token here

File: api/controllers/CategoryController.js (hunk 1 above)
Comment (josemak25) [reply api/controllers/CategoryController.js]: ok, sir fixed that.

File: api/middlewares/IsAdmin.js (hunk 2 above)
Comment (josemak25) [reply api/middlewares/IsAdmin.js]: made changes to middleware response handler

File: api/middlewares/IsAdmin.js (hunk 3 above)
Comment (josemak25) [reply api/middlewares/IsAdmin.js]: made changes to the middleware response handler

File: api/middlewares/IsInvestor.js (hunk 4 above)
Comment (josemak25) [reply api/middlewares/IsInvestor.js]: made changes to middleware response handler

File: api/queries/category.queries.js (hunk 5 above)
Comment (josemak25) [reply api/queries/category.queries.js]: changed categories to category

File: test/controllers/CategoryController.test.js (hunk 7 above)
Comment (josemak25) [reply test/controllers/CategoryController.test.js]: can't use token service to generate the token cause on verification it returns null and not the object, both the login flow works

**Pull request merged**